### PR TITLE
Remove ksonnet from ingress.jsonnet

### PR DESCRIPTION
**Pull request by kakkoyun**
Signed-off-by: Kemal Akkoyun <kakkoyun@gmail.com>

Tested with:
* `jsonnet -J vendor -m manifests examples/ingress.jsonnet | xargs -I{} sh -c 'cat {} | gojsontoyaml > {}.yaml' -- {}`
* `git diff new_manifests old_manifets`

### CODE REVIEW

File: docs/exposing-prometheus-alertmanager-grafana-ingress.md [hunk 1]
```diff
@@ -48,30 +41,44 @@ local kp =
...
-          ingressRule.new() +
-          ingressRule.withHost('prometheus.example.com') +
-          ingressRule.mixin.http.withPaths(
-            httpIngressPath.new() +
-            httpIngressPath.mixin.backend.withServiceName($.prometheus.service.metadata.name) +
-            httpIngressPath.mixin.backend.withServicePort('web')
-          ),
-        ),
+      'prometheus-k8s': {
+        apiVersion: 'extensions/v1beta1',
```
Comment (lilic) [inline docs/exposing-prometheus-alertmanager-grafana-ingress.md:45]: Actually this should be `networking.k8s.io/v1` as `extensions/v1beta1 ` is being deprecated so best to use latest API in the examples.

File: examples/ingress.jsonnet [hunk 2]
```diff
@@ -1,9 +1,17 @@
-local k = import 'ksonnet/ksonnet.beta.3/k.libsonnet';
-local secret = k.core.v1.secret;
-local ingress = k.extensions.v1beta1.ingress;
-local ingressTls = ingress.mixin.spec.tlsType;
-local ingressRule = ingress.mixin.spec.rulesType;
-local httpIngressPath = ingressRule.mixin.http.pathsType;
+local ingress(name, namespace, rules) = {
+  apiVersion: 'extensions/v1beta1',
```
Comment (lilic) [inline examples/ingress.jsonnet:2]: Same as above.

**Review by lilic — COMMENTED** [review]
Thanks! One comment regarding upgrading API. Note that Ingress changed a couple of fields that are not required. https://kubernetes.io/docs/concepts/services-networking/ingress/#the-ingress-resource

**Comment by kakkoyun** [comment]
Thanks for the review @lilic. I kept them as is, hence the deprecated API. I'll be also updating them from now on.a

**Review by lilic — APPROVED** [review]
lgtm

**Pull request merged**